### ISSUE TITLE
TFRecords writing and reading

### ISSUE THREAD

**Issue by NaveedMazhar-eng** [tensorflow/tensorflow#45101]
Hi,
i have a very huge data and which I want to convert to tfrecords before sending it to neural network. My writing code is 
`for n, pat in tqdm(enumerate(pats)):
    cache_file = cache_dir+f'{pat}.tfrecords'
    writer = tf.io.TFRecordWriter(cache_file)
    files = os.listdir(pat_dir+pat+'/')
    files.sort()
    X_train = []
    for i, file in enumerate(files):
        byte_image = tf.io.read_file(pat_dir+pat+'/'+file)
        image = tfio.image.decode_dicom_image(byte_image, dtype=tf.uint16)
        image = image.numpy()
        image = image/(image.max())
        X_train.append(image)
    X_train = np.array(X_train).reshape((180, 256, 256, 1))
    Y_train = imread(lab_dir+lab_files[n], as_gray=True)
    Y_train = tf.keras.utils.to_categorical(Y_train, num_classes=3)
    # plt.imshow(Y_train)
    # plt.show()
    feature = { 'train/label'  : _bytes_feature(tf.compat.as_bytes(Y_train.tostring())),
                 'train/images' : _bytes_feature(tf.compat.as_bytes(X_train.tostring()))
               }
    example = tf.train.Example(features=tf.train.Features(feature=feature))
    writer.write(example.SerializeToString())
    writer.close()
    del writer
    gc.collect()`
which i think is working perfectly fine but when i follow the example given on keras page to read tfrecords [https://keras.io/examples/keras_recipes/tfrecord/](url) it reads all the files fine but for last two files it gives a dimension error. My reading code is 
`def read_tfrecord(example):
    tfrecord_format ={'train/label'  : tf.io.FixedLenFeature([], tf.string),
            'train/images' : tf.io.FixedLenFeature([], tf.string)}
    example = tf.io.parse_single_example(example, tfrecord_format)
    tr_images = tf.io.decode_raw(example['train/images'], tf.float32)
    tr_label = tf.io.decode_raw(example['train/label'], tf.float32)
    tr_images = tf.reshape(tr_images, [180, 256, 256, 1])
    tr_label = tf.reshape(tr_label, [1, 256, 256, 3])
    return tr_images, tr_label`
the problem is that it reads all the fines just fine but last two files are merged together and when gives this error
`Cannot reshape a tensor with 2949120 elements to shape [180,256,256,1] (11796480 elements) for '{{node Reshape}} = Reshape[T=DT_FLOAT, Tshape=DT_INT32](DecodePaddedRaw, Reshape/shape)' with input shapes: [2949120], [4] and with input tensors computed as partial shapes: input[1] = [180,256,256,1].`
please help!
sorry for my English it's not good but i hope that someone gets my point and will reply me soon. and also this is my 1st time with any issue so forgive me if i have not follow the pattern.

**Comment by hfahrudin** [comment]
@NaveedMazhar-eng could you please provide your system information? Also, it would be great if you show your code snippets nicely. Thank you!

**Comment by Saduf2019** [comment]
@NaveedMazhar-eng 
We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code to reproduce the issue faced]
